api/source/graphql/mutations/ApplicationMutation.ts

```diff
@@ -0,0 +1,176 @@
+
+import { ApplicationType } from "../../types/ApplicationType";
+import { GraphQLBoolean, GraphQLError, GraphQLID, GraphQLNonNull, GraphQLString } from "graphql";
+import { v4 as uuid } from "uuid";
+import {
+  APPLICATION_STATUS_TABLE_NAME,
+  APPLICATION_TABLE_NAME,
+  CITATION_TABLE_NAME,
+  USER_TABLE_NAME,
+} from "../../database/utils/database_constants";
+import { db } from "../../database/database";
+
+export default {
+  createApplication: {
+    type: ApplicationType,
+    args: {
+      user_id: {
+        type: GraphQLNonNull(GraphQLID),
+      },
+      title: {
+        type: GraphQLNonNull(GraphQLString),
+      },
+      deadline: {
+        type: GraphQLString,
+      },
+      end_time: {
+        type: GraphQLString,
+      },
+      emission_date: {
+        type: GraphQLNonNull(GraphQLString),
+      },
+      response_date: {
+        type: GraphQLNonNull(GraphQLString),
+      },
+      application_status_id: {
+        type: GraphQLNonNull(GraphQLID),
+      },
+      citation_id: {
+        type: GraphQLNonNull(GraphQLID),
+      },
+    },
+    resolve: async (
+      _: any,
+      {
+        user_id,
+        title,
+        deadline,
+        start_time,
+        end_time,
+        emission_date,
+        response_date,
+        application_status_id,
+        citation_id,
+      }: any,
+    ) => {
+      const id = uuid();
+
+      const newApplication = await db
+        .select()
+        .from(APPLICATION_TABLE_NAME)
+        .where("id", id)
+        .catch((error: Error) => {
+          console.error(error);
+          throw new GraphQLError(error.name);
+        });
+
+      const myUser = await db
+        .select()
+        .from(USER_TABLE_NAME)
+        .where({ id: user_id })
+        .catch((error: Error) => {
+          console.error(error);
+          throw new GraphQLError(error.name);
+        });
+
+      const myApplytatus = await db
+        .select()
+        .from(APPLICATION_STATUS_TABLE_NAME)
+        .where({ id: application_status_id })
+        .catch((error: Error) => {
+          console.error(error);
+          throw new GraphQLError(error.name);
+        });
+
+      const myCitation = await db
+        .select()
+        .from(CITATION_TABLE_NAME)
+        .where({ id: citation_id })
+        .catch((error: Error) => {
+          console.error(error);
+          throw new GraphQLError(error.name);
+        });
+
+      await db(APPLICATION_TABLE_NAME)
+        .insert({
+          id,
+          title,
+          deadline,
+          start_time,
+          end_time,
+          emission_date,
+          response_date,
+          user_id,
+          application_status_id,
+          citation_id,
+        })
+        .catch((error: Error) => {
+          console.error(error);
+          throw new GraphQLError(error.name);
+        });
+
+      return {
+        ...newApplication[0],
+        user: myUser[0],
+        applicationStatus: myApplytatus[0],
```

Comment:
This mutation is wrong, you are fetching an `nonexistent` application and then inserting a new one. The first query 
```
const newApplication = await db
        .select()
        .from(APPLICATION_TABLE_NAME)
        .where("id", id)
        .catch((error: Error) => {
          console.error(error);
          throw new GraphQLError(error.name);
        });
```

will return an empty array because it is trying to find an `Application` with the random generated ID.

That fetch must be executed once the new application has been inserted into the database.